robots/cmd/test-label-analyzer/cmd/root.go

```diff
@@ -98,34 +106,70 @@ func (s *configOptions) validate() error {
 }
 
 // getConfig returns a configuration with which the matching tests are being retrieved or an error in case the configuration is wrong
-func (s *configOptions) getConfig() (*testlabelanalyzer.Config, error) {
+func (s *ConfigOptions) getConfig() (*testlabelanalyzer.Config, error) {
```

Comment:
TBH, coming from the Java world, I still struggle with that pattern. But you are right, it makes it more clear to read. However, I'll also extract some methods to avoid it being too long.